include/alpaka/dev/DevCpu.hpp

```diff
@@ -218,6 +218,26 @@ namespace alpaka
                 }
             };
 
+            //#############################################################################
+            //! The CPU device available shared memory per block get trait specialization.
+            //!
+            //! Currently implementation for CPUs is a stub that treats all memory
+            //! as available as shared memory per block
+            template<>
+            struct GetSharedMemBytes<
+                dev::DevCpu>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto getSharedMemBytes(
+                    dev::DevCpu const & dev)
+                -> std::size_t
+                {
+                    alpaka::ignore_unused(dev);
+
+                    return dev::cpu::detail::getTotalGlobalMemSizeBytes();
```

Comment:
Only the accelerator knows the amount of available shared memory. 
After looking again into PR #1033 and the help of @sbastrakov we saw that this fixed size shared memory is only used for dynamic shared memory.

This will now race the question what does this trait should give back. IMO the alpaka concept has only one shared memory which can be used via static and dynamic allocations.

Reply:
Agree, will modify the PR to accomplish it.

Reply:
Fixed and rebased.